cf/env/env_test.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package env
+
+import (
+	"errors"
+	"os"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestApi(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "CF Env Suite")
+}
+
+var _ = Describe("CF Env", func() {
+
+	BeforeSuite(func() {
+		os.Setenv("VCAP_APPLICATION", "{}")
```

Comment:
modifying os env vars in tests is normally not advised - it may cause issues when tests are ran concurrently. It may also cause issues if other processes running currently depend on this env variables.

However I don't have any great ideas of how we can test the actual logic.
Option one would be to move cfenv.Current in main and pass cf.App to our cfenv. I don't know whether this makes sense or not though.
Option two would be to stub the os enivonment in the test. For this to work we would need to slightly wrap the os environment and inject the wrapper in our cf env (so that we can later stub that with some fake env implementation of the stub wrapper). 

Personally, I dont like either of these ideas. but I am not sure its good to touch os envs in tests either.

Reply:
Are we going to run tests concurrently in the same process? Here the environment is cleared in After* so other tests are not affected. Wrap the environment just to test it in some other way seems too much imo.